File: stacks_queues/stacks.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    stack<int> s1;
    stack<string> s2;

    s1.push(1);
    s1.push(3);
    s1.push(7);

    for (int i = 4; i < 7; i++)
        s1.push(i);         // I am pushing 4, 5, 6

    // Total 6 elements in s1: Order of push: 1 3 7 4 5 6

    while (!s1.empty()) {
        int temp = s1.top();
        cout <<temp << " ";
        s1.pop();
    }
    cout << endl;

    s2.push("sagar");
    s2.push("vishnu");
    s2.push("shreya");

    while (!s2.empty()) {
        string temp = s2.top();
        cout << temp << " ";
        s2.pop();
    }
    cout << endl;

    return 0;
}

File: stacks_queues/stacks.py
s1 = []
s2 = []

s1.append(1)
s1.append(3)
s1.append(7)

for i in range(4, 7):
    s1.append(i)  # Pushing 4, 5, 6

# Total 6 elements in s1: Order of push: 1 3 7 4 5 6

while s1:
    temp = s1.pop()
    print(temp, end=" ")
print()

s2.append("sagar")
s2.append("vishnu")
s2.append("shreya")

while s2:
    temp = s2.pop()
    print(temp, end=" ")
print()
